python/lsst/cp/pipe/ptc/cpExtractPtcTask.py

```diff
@@ -792,40 +843,35 @@ def getGainFromFlatPair(self, im1Area, im2Area, imStatsCtrl, mu1, mu2,
 
         return gain
 
-    def getReadNoiseFromMetadata(self, taskMetadata, ampName):
+    def getReadNoise(self, exposure, taskMetadata, ampName):
         """Gets readout noise for an amp from ISR metadata.
 
         Parameters
         ----------
-        taskMetadata : `list` [`lsst.pipe.base.TaskMetadata`]
-                    List of exposures metadata from ISR.
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to check for read noise first.
+        taskMetadata : `lsst.pipe.base.TaskMetadata`
+            List of exposures metadata from ISR for this exposure.
         ampName : `str`
             Amplifier name.
 
         Returns
         -------
         readNoise : `float`
-            Median of the overscan readnoise in the
-            post-ISR metadata of the input exposures (ADU).
-            Returns 'None' if the median could not be calculated.
+            The read noise for this set of exposure/amplifier.
         """
-        # Empirical readout noise [ADU] measured from an
-        # overscan-subtracted overscan during ISR.
-        expectedKey = f"RESIDUAL STDEV {ampName}"
+        # Try from the exposure first.
+        expectedKey = f"LSST ISR OVERSCAN RESIDUAL SERIAL STDEV {ampName}"
```

Comment:
Hoes does this noise from the exposure metadata potentially differ from the noise form the task metadata? And why did you need to add this extra check?

Reply:
They should be identical.  I added this because I was finding that all my read noise values were being set to NaN.  The `cp_testing` pipeline renames the `IsrTask` stage of the pipeline to `cptPtcIsr` to make it unique within the pipeline, and that changes the key within the `taskMetadata` object that stores that information.  I originally tried making that key configurable, but decided that that was a work-around to a work-around, so I added the lookup in the exposure metadata instead.  As this header key should be present in all ISR processed exposures from now on, this change is also the first step to deprecating and removing the `taskMetadata` lookup and connection entirely.

Reply:
"is also the first step to deprecating and removing the taskMetadata lookup and connection entirely"--> :) :)